lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -90,89 +104,221 @@ impl EnsoLexer {
-    /// important so as not to result in incorrect spans on windows clients.
-    fn newline() -> Pattern {
-        Pattern::char('\n') | Pattern::all_of("\r\n")
+    /// Triggered when a line is ended in a doc comment.
+    fn on_doc_comment_end_of_line<R:ReaderOps>
+    ( &mut self
+    , line_ending:token::LineEnding
+    , reader:&mut R
+    ) {
+        trace!(self.logger,"Comment::on_doc_comment_end_of_line");
+        self.comment_state.submit_line(line_ending);
+        match line_ending {
+            token::LineEnding::None => {
+                let matched_bookmark = self.bookmarks.matched_bookmark;
+                self.bookmarks.rewind(matched_bookmark,reader);
+                self.on_end_doc_comment(reader);
+                self.on_eof(reader);
+            },
+            _ => {
+                let comment_newline = self.doc_comment_newline;
+                self.push_state(comment_newline);
+            }
+        }
     }
 
-    /// The allowable group characters in Enso.
-    fn group_chars() -> String {
-        String::from("()[]{}")
+    /// Triggered when ending a doc comment.
+    fn on_end_doc_comment<R:ReaderOps>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Comment::on_end_doc_comment");
+        let doc_comment = self.doc_comment;
+        let mut comment = self.comment_state.consume_current();
+        let blank_lines = comment.consume_blank_lines();
+        if !blank_lines.is_empty() {
+            self.block_state.seen_newline = true;
+        }
+        self.block_state.delayed_append_lines.extend(blank_lines);
+        let comment_token = comment.into();
+        self.append_token(comment_token);
+        self.pop_states_including(doc_comment);
     }
 
-    /// The allowable operator characters in Enso.
-    fn operator_chars() -> String {
-        String::from(";!$%&*+-/<>?^~|:\\")
+    /// Triggered when a new line is discovered in a doc comment.
+    fn doc_comment_on_new_line<R:ReaderOps>(&mut self, reader:&mut R) {
+        trace!(self.logger,"Comment::doc_comment_on_new_line");
+        self.pop_state();
+        let indent         = self.consume_current().chars().count();
+        let comment_indent = self.comment_state.current_comment.indent;
+        if indent < comment_indent {
+            self.on_end_doc_comment(reader);
+            self.block_on_newline(token::LineEnding::None,reader);
+        } else if (indent - comment_indent) > 0 {
+            let remaining_indent = lexeme::space().literal().repeat(indent - comment_indent);
+            self.comment_state.append_to_line(remaining_indent);
+        }
     }
 
-    /// The characters that break tokens in Enso.
-    fn whitespace_break_chars() -> String {
-        String::from("\t\r\n")
+    /// Triggered when an empty line is discovered in a doc comment.
+    fn doc_comment_on_empty_line<R:ReaderOps>
+    ( &mut self
+    , line_ending:token::LineEnding
+    , reader:&mut R
+    ) {
```

Comment:
according to style guide, we align `:` here and put spaces around it, so it should be this:
```rust
    fn doc_comment_on_empty_line<R:ReaderOps>
    ( &mut self
    , line_ending : token::LineEnding
    , reader      : &mut R
    ) {
```